violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/AddNodeBehavior.java

```diff
@@ -17,33 +15,62 @@
 import com.horstmann.violet.workspace.sidebar.graphtools.IGraphToolsBar;
 import com.horstmann.violet.workspace.sidebar.graphtools.IGraphToolsBarMouseListener;
 
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Point2D;
+import java.util.Collection;
+
+import static javax.swing.JOptionPane.showMessageDialog;
+
 public class AddNodeBehavior extends AbstractEditorPartBehavior implements IGraphToolsBarMouseListener
 {
-	private static final int OUTSIDE_SCREEN_POSITION = -1000;
+    /** default number of max initial nodes on workspace */
+    private static final int maxNumberOfInitialNodes = 1;
+    /** default screen position*/
+    private static final int OUTSIDE_SCREEN_POSITION = -1000;
+
+    private IEditorPart editorPart;
+
+    private IGraph graph;
+
+    private IEditorPartSelectionHandler selectionHandler;
+
+    private IEditorPartBehaviorManager behaviorManager;
+
+    private IGraphToolsBar graphToolsBar;
+
+    private boolean dragging;
+
+    private INode draggedNode;
+
+    @ResourceBundleBean(key = "edit.properties.empty_message")
+    private String message;
 
     public AddNodeBehavior(IEditorPart editorPart, IGraphToolsBar graphToolsBar)
     {
-    	this.editorPart = editorPart;
+        ResourceBundleInjector.getInjector().inject(this);
+        this.editorPart = editorPart;
         this.graph = editorPart.getGraph();
         this.selectionHandler = editorPart.getSelectionHandler();
         this.behaviorManager = editorPart.getBehaviorManager();
         this.graphToolsBar = graphToolsBar;
-        
         this.dragging = false;
         this.draggedNode = null;
         graphToolsBar.addMouseListener(this);
     }
-    
+
     @Override
     public void onMouseClicked(MouseEvent event)
     {
-    	this.draggedNode = null;
-    	
+        this.draggedNode = null;
+
         if (event.getClickCount() > 1)
         {
             return;
```

Comment:
Maybe only if(x && y) { 
//action
} 
instead of this? It's only a suggestion.